examples/point_cloud/main.cpp

```diff
@@ -131,15 +131,15 @@ int main(int argc, char *argv[]) {
     }
 
     /* ccb calibration_file*/
-    err = snprintf(ccb_filename, sizeof(ccb_filename), "%s",
+    err = snprintf(const_cast<char*>(ccb_filename.c_str()), sizeof(ccb_filename.c_str()), "%s",
```

Comment:
I don't think this operation is allowed. The internal char array of a string shouldn't be modified like this. The c_str() method returns a constant char array for a reason.
If you need to concatenate strings, or construct a string from other strings, it can be done easily, just check the documentation.
Also, please build and test to make sure the sdk or the examples functionality is the same. This PR is about removing warnings, so the functionality should not be affected.

Reply:
Solved